src/client/dfs/dfs.c

```diff
@@ -2431,14 +2445,16 @@ dfs_osetattr(dfs_t *dfs, dfs_obj_t *obj, struct stat *stbuf, int flags)
 	rc = daos_obj_update(oh, th, &dkey, akeys_nr, iods, sgls, NULL);
 	if (rc) {
 		D_ERROR("Failed to update attr (rc = %d)\n", rc);
-		D_GOTO(out, rc = -daos_der2errno(rc));
+		D_GOTO(err, rc = daos_der2errno(rc));
 	}
 
+out:
+
 	rc = entry_stat(dfs, th, oh, obj->name, stbuf);
```

Comment:
we can optimize this if the user is setting the size on a file, since we don't need to get the file size in the stat anymore. set/get size is a little expensive and it seems we can avoid a get size in the stat if we have set the size already. it's fine to leave it to a different patch though.